esp-hal/src/macros.rs

```diff
@@ -168,3 +168,71 @@ macro_rules! if_set {
 macro_rules! ignore {
     ($($item:tt)*) => {};
 }
+
+/// Macro to define the application descriptor.
+#[macro_export]
+macro_rules! esp_app_desc {
```

Comment:
This does feel a bit odd living here in the HAL, but I'm not sure where else it should go either 🤔. I suppose we should also mark it as unstable for now?

Reply:
t.b.h. I am not sure I like to have this as a macro at all

My other idea would be to generate the app-desc in build.rs and let the linker include it - but then we would need to get the app-name and app-version from a config option instead (or play some dirty tricks to get the binary's manifest)